dissect/target/plugins/os/unix/bsd/osx/network.py

```diff
@@ -0,0 +1,91 @@
+from __future__ import annotations
+
+import plistlib
+from functools import lru_cache
+from typing import Iterator
+
+from dissect.target.helpers.record import MacInterfaceRecord
+from dissect.target.plugins.general.network import NetworkPlugin
+from dissect.target.target import Target
+
+
+class MacNetworkPlugin(NetworkPlugin):
+    def __init__(self, target: Target):
+        super().__init__(target)
+        self._plistlease = lru_cache(4096)(self._plistlease)
```

Comment:
```suggestion
        self._plistlease = lru_cache(32)(self._plistlease)
```
Smaller cache size is more fitting.